lib/src/singleton.dart

```diff
@@ -4,5 +4,20 @@
 library grinder.src.singleton;
 
 import '../grinder.dart';
+import 'utils.dart';
 
 final Grinder grinder = new Grinder();
+
+final ZonedValue zonedContext = new ZonedValue(new _NoopContext());
+
+class _NoopContext implements GrinderContext {
```

Comment:
If we create new GrinderContexts for each task / zone then it will be possible to use the GrinderContext outside of the associated task execution window.  Do we want to allow that?  If no, what if we have GrinderContext be a singleton (still accessed via `context`) and then we just store the task in each zone:

```dart
final GrinderContext context = new GrinderContext();

class GrinderContext {
  GrinderTask get task => zonedTask.value;
  Grinder get grinder => grinder; // from singleton.dart
  void log(String msg) {
    if (task == null) throw new GrinderException('GrinderContext.log can only be called during execution of a task');
    // log ...
  }

  // similar for fail ...
}
```